app/Http/Controllers/Shopify/AppController.php

```diff
@@ -225,158 +225,162 @@ public function printLabels(Request $request) {
                 continue;
             }
 
-            $done_shipment = ShopifyShipment::where('shop_id', $shop->id)
-                    ->where('order_id', $order['legacyResourceId'])
-                    ->where('test_mode', $shop->test_mode)
-                    ->where('return', $is_return)
-                    ->first();
+            DB::transaction(function () {
 
-            if ($done_shipment) {
-                $shipment['status'] = 'sent';
+                $done_shipment = ShopifyShipment::where('shop_id', $shop->id)
+                        ->where('order_id', $order['legacyResourceId'])
+                        ->where('test_mode', $shop->test_mode)
+                        ->where('return', $is_return)
+                        ->first();
 
-                if (strpos($done_shipment->tracking_code, ',')) {
-                    $tracking_codes = explode(', ', $done_shipment->tracking_code);
-                } else {
-                    $tracking_codes[] = $done_shipment->tracking_code;
+                if ($done_shipment) {
+                    $shipment['status'] = 'sent';
+
+                    if (strpos($done_shipment->tracking_code, ',')) {
+                        $tracking_codes = explode(', ', $done_shipment->tracking_code);
+                    } else {
+                        $tracking_codes[] = $done_shipment->tracking_code;
+                    }
+                    $shipment['tracking_codes'] = $tracking_codes;
+                    $shipments[] = $shipment;
+                    continue;
```

Comment:
Can't call continue from DB::transaction(function() {}); - has to be handled some other way